### PR TITLE
Add symfony 3 compatibility

### PR DESCRIPTION
**Pull request by alexander-schranz**
# TODO 

 - [x] UPGRADE.md
 - [x] CHANGELOG.md
 - [x] Dynamic Media Upload
 - [x] Static Form Loading (configuration mapping templateKey to class)
 - [x] Email Handling
   - [x] Static Mail Attributes
   - [x] Dynamic Mail Attributes
   - [x] Remove MailSubscriber
 - [x] Add Alternative to DynFormSavedEvent

### CODE REVIEW

File: Controller/FormWebsiteController.php [hunk 1]
```diff
@@ -38,42 +39,54 @@ class FormWebsiteController extends DefaultController
...
-                        400
-                    );
-                }
             }
-        }
 
-        $this->createFormBuilder();
-
-        $response = parent::indexAction($structure, $preview, $partial);
+            if ($request->isXmlHttpRequest()) {
```
Comment (wachterjohannes) [inline Controller/FormWebsiteController.php]: no nested if

File: Event/RequestListener.php [hunk 2]
```diff
@@ -90,22 +97,16 @@ public function onKernelRequest(GetResponseEvent $event)
...
-            );
-
-            $dynFormSavedEvent = new DynFormSavedEvent($serializedObject, $dynamic);
-            $this->eventDispatcher->dispatch(DynFormSavedEvent::NAME, $dynFormSavedEvent);
-
-            $response = new RedirectResponse('?send=true');
-            $event->setResponse($response);
+            $configuration = $this->formConfigurationFactory->buildByDynamic($dynamic);
+
+            if ($this->formHandler->handle($form, $configuration)) {
```
Comment (wachterjohannes) [inline Event/RequestListener.php]: try to avoid nested if

File: Event/RequestListener.php [hunk 3]
```diff
@@ -62,45 +69,36 @@ public function __construct(
...
-                ]
-            );
+        /** @var Dynamic $dynamic */
+        $dynamic = $form->getData();
+        $configuration = $this->formConfigurationFactory->buildByDynamic($dynamic);
 
+        if ($this->formHandler->handle($form, $configuration)) {
+            $serializedObject = $dynamic->getForm()->serializeForLocale($dynamic->getLocale(), $dynamic);
             $dynFormSavedEvent = new DynFormSavedEvent($serializedObject, $dynamic);
             $this->eventDispatcher->dispatch(DynFormSavedEvent::NAME, $dynFormSavedEvent);
```
Comment (matt-halliday) [inline Event/RequestListener.php]: The `DynFormSavedEvent` is marked as deprecated. The MailChimpListener is subscribed to it. Should this be refactored to use `FormEvent`?

Comment (alexander-schranz) [reply Event/RequestListener.php]: Thx did fix that.

**Comment by alexander-schranz** [comment]
@wachterjohannes for me it seems all is working as expected now.

**Comment by alexander-schranz** [comment]
@matt-halliday release is tagged

**Pull request merged**